charts/gardener-extension-provider-alicloud/templates/service.yaml

```diff
@@ -3,8 +3,13 @@ kind: Service
 metadata:
   name: {{ include "name" . }}
   namespace: {{ .Release.Namespace }}
-{{-  if .Values.ignoreResources }}
   annotations:
+    networking.resources.gardener.cloud/from-world-to-ports: '[{"protocol":"TCP","port":{{ .Values.webhookConfig.serverPort }}}]'
+    networking.resources.gardener.cloud/from-policy-pod-label-selector: all-seed-scrape-targets
+    networking.resources.gardener.cloud/from-policy-allowed-ports: '[{"port":{{ .Values.metricsPort }},"protocol":"TCP"}]'
+    networking.resources.gardener.cloud/namespace-selectors: '[{"matchLabels":{"kubernetes.io/metadata.name":"garden"}}]'
+    networking.resources.gardener.cloud/pod-label-selector-namespace-alias: extensions
+{{- if .Values.ignoreResources }}
```

Comment:
About mutations from provider extension in seed cluster and in shoot cluster, how to enable seed api server pod and shoot api server pod to communicate to provider extension pod?

Reply:
Seed API server -> provider extension: https://github.com/gardener/gardener-extension-provider-alicloud/pull/581/commits/80bc4be9234621be84c0e28b1a057309259e4a0d#diff-7cb3f66c69cb7755749be614f2ffa16a773b918523ac7df9652b457185059e22R7
Shoot API server -> provider extension: https://github.com/gardener/gardener-extension-provider-alicloud/pull/581/commits/3f1780cc1f51c57f83007de521839bb003b4b296#diff-cc0552e76e65ffff14fa4d53e143f0d325fb62ff3c926693a9153658dcbfd23dR55

Reply:
With regards to "Seed API server -> provider extension", is it possible to restrict network policy instead of "from-world-to-ports"? 
about "Shoot API server -> provider extension",  the commit you mentioned is for csi validation. what I mean is gardener-extension-provider-alicloud-shoot which calls "https://gardener-extension-provider-alicloud.extension-provider-alicloud-bmklt:443/shoot"

Reply:
> 
With regards to "Seed API server -> provider extension", is it possible to restrict network policy instead of "from-world-to-ports"?

What do you want to restrict? With this policy, the access is only allowed to the webhook server port. If you want to restrict it further then you need to create your own custom `NetworkPolicy`. However, before this PR, there were no `NetworkPolicy`s at all for namespaces, so it's not getting worse now 😉 

> about "Shoot API server -> provider extension", the commit you mentioned is for csi validation. what I mean is gardener-extension-provider-alicloud-shoot which calls "[https://gardener-extension-provider-alicloud.extension-provider-alicloud-bmklt:443/shoot](https://gardener-extension-provider-alicloud.extension-provider-alicloud-bmklt/shoot)"

This network path is ensured by a custom `NetworkPolicy` created by the extensions library (this was already the case before this PR): https://github.com/gardener/gardener/blob/master/extensions/pkg/webhook/shoot/networkpolicy.go

Reply:
"Shoot API server -> provider extension". --> got it. I thought all network policies will be managed via new way. I have no question if custom policy is still allowed for this case.
"Seed API server -> provider extension" --> ok. from extension service perspective, all ingress from any IP is allowed to webhook server port. from seed api pod perspective, I see a label in it "networking.gardener.cloud/to-private-networks: allowed", which I guess enable egress from seed API pod to extension service.  Then It should work. No question anymore about it.
> > 
> 
> With regards to "Seed API server -> provider extension", is it possible to restrict network policy instead of "from-world-to-ports"?
> 
> What do you want to restrict? With this policy, the access is only allowed to the webhook server port. If you want to restrict it further then you need to create your own custom `NetworkPolicy`. However, before this PR, there were no `NetworkPolicy`s at all for namespaces, so it's not getting worse now 😉
> 
> > about "Shoot API server -> provider extension", the commit you mentioned is for csi validation. what I mean is gardener-extension-provider-alicloud-shoot which calls "[https://gardener-extension-provider-alicloud.extension-provider-alicloud-bmklt:443/shoot](https://gardener-extension-provider-alicloud.extension-provider-alicloud-bmklt/shoot)"
> 
> This network path is ensured by a custom `NetworkPolicy` created by the extensions library (this was already the case before this PR): https://github.com/gardener/gardener/blob/master/extensions/pkg/webhook/shoot/networkpolicy.go